### PR TITLE
docs: add "Updating Your Branch" in contributing/workflow.md

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
The step after pushing PR is missing.

**Checklist:**
- [x] Securely signed commits
- [] Component(s) with PHPDoc blocks, only if necessary or adds value
- [] Unit testing, with >80% coverage
- [] User guide updated
- [] Conforms to style guide

### CODE REVIEW

File: contributing/workflow.md [hunk 1]
```diff
@@ -215,6 +214,33 @@ Label your PRs with the one of the following [labels](https://github.com/codeign
...
+## Updating Your Branch
+
+If you are asked for changes in the review, commit the fix in your branch and push it to GitHub again.
+
+If the `develop` branch progresses and conflicts arise that prevent merging, or if you are asked to *rebase*,
+do the following:
+
+Synchronize your repository:
+
+    git checkout develop
```
Comment (MGatner) [inline contributing/workflow.md:226]: Let's encourage the modern Guy commands:
```suggestion
    git switch develop
```

File: contributing/workflow.md [hunk 2]
```diff
@@ -215,6 +214,33 @@ Label your PRs with the one of the following [labels](https://github.com/codeign
...
+    git fetch upstream
+    git merge upstream/develop
+    git push origin develop
```
Comment (MGatner) [inline contributing/workflow.md:229]: I prefer to keep it simple, unless you think the verbosity adds something.
```suggestion
    git pull upstream develop
    git push
```

File: contributing/workflow.md [hunk 3]
```diff
@@ -215,6 +214,33 @@ Label your PRs with the one of the following [labels](https://github.com/codeign
...
+
+Bring your feature branch up to date:
+
+    git checkout new/mind-reader
```
Comment (MGatner) [inline contributing/workflow.md:233]:
```suggestion
    git switch new/mind-reader
```

**Review by MGatner — CHANGES_REQUESTED** [review]

File: contributing/workflow.md (hunk 2 above)
Comment (kenjis) [reply contributing/workflow.md:229]: I strongly oppose it. I think the verbosity adds values.
This document is for git beginners. 
Omitted commands hide what they are doing.

`git pull` is harmful.
https://stackoverflow.com/questions/15316601/in-what-cases-could-git-pull-be-harmful

`git push` does not show where and what you push.

File: contributing/workflow.md (hunk 1 above)
Comment (kenjis) [reply contributing/workflow.md:226]: Is anyone using a lower version that doesn't allow git switch anymore?

Comment (datamweb) [reply contributing/workflow.md:226]: I use `git switch my-branch` and find it more readable.

File: contributing/workflow.md (hunk 2 above)
Comment (datamweb) [reply contributing/workflow.md:229]: @MGatner I've been using git lately. I used to the desktop version of GitHub. In the desktop version, I used pull. `Pull` created  problems for me. I am generally opposed to this change.

File: contributing/workflow.md (hunk 1 above)
Comment (MGatner) [reply contributing/workflow.md:226]: It's almost three years old now, and 13 versions behind the current. There may still be people using older versions (I think Ubuntu 18.04 still ships lower?) but I would prefer for us to support the new nomenclature. If someone needs this tutorial they are likely on a fairly new install of Git.

Comment (kenjis) [reply contributing/workflow.md]: Okay, I updated. https://github.com/codeigniter4/CodeIgniter4/pull/6045/commits/dc3896d4d247117bc9643694227090b33dff3716

File: contributing/workflow.md (hunk 3 above)
Comment (datamweb) [reply contributing/workflow.md]: 👍

**Review by paulbalandan — APPROVED** [review]

**Pull request merged**